### ISSUE TITLE
Incorrect `Instance of 'object' has no 'bar' member` error when using `OrderedDict`s

### ISSUE THREAD

**Issue by timvandermeij** [PyCQA/pylint#1119]
### Steps to reproduce

Run the following minimal testcase:

``` python
from collections import OrderedDict

class Foo(object):
    def bar(self):
        return "baz"

a = OrderedDict()
a["foo"] = Foo()
b = a.popitem(last=True)[1]
print(b.bar())
```
### Current behavior

Notice that Pylint raises the following error: `Instance of 'object' has no 'bar' member`
### Expected behavior

Pylint recognizes that the object returned from the `OrderedDict` is a `Foo` object and it thus has the `bar` method. Note that this does work with the following code that just uses a regular dictionary:

```
class Foo(object):
    def bar(self):
        return "baz"

a = {}
a["foo"] = Foo()
b = a.popitem()[1]
print(b.bar())
```
### pylint --version output

```
No config file found, using default configuration
pylint 1.6.4, 
astroid 1.4.8
Python 2.7.11 (default, Mar 31 2016, 06:18:34) 
[GCC 5.3.0]
```

**Comment by timvandermeij** [comment]
Thank you for fixing this!
